### PR TITLE
add task solution React_clock

### PR DESCRIPTION
**Pull request by TamaraShukalo**
[DEMO LINK](https://TamaraShukalo.github.io/react_clock/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,55 @@
...
+    }, 3300);
+
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.hideClock);
+    document.removeEventListener('click', this.showClock);
+  }
```
Comment (qoa11a) [inline src/App.tsx:34]: You need to clear intervals when unmounting a component.

File: src/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+type State = {
+  today: Date;
+};
+
+export class Clock extends React.Component<Props> {
+  state: State = {
+    today: new Date(),
+  };
+
+  timerId = 0;
```
Comment (qoa11a) [inline src/Clock.tsx:17]:
```suggestion
  intervalId = 0;
```
I think that `timerId` isn't aptly named.

**Review by qoa11a — CHANGES_REQUESTED** [review]
Good job! 👍 
A few suggestions.

**Review by YegorKorenkov — APPROVED** [review]
Good job!

**Pull request closed without merging**